src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
I believe that if you change the assembly helper to use ESP frame instead of EBP one, you would not have this problem. It seems all you'd need to do is to remove the PROLOG_BEG / PROLOG_END and EPILOG_BEG / EPILOG_END (or replace them with dummy ESP_PROLOG_BEG, ... to make it kind of symmetric with EBP based frames) and add `PROLOG_PUSH ebp` at the beginning and `EPILOG_POP ebp` at the end. Then changing the EBP in the middle of the function won't cause trouble, since the unwinder would use ESP for unwinding.